apps/frontend/src/components/RequestModal/RequestModal.tsx

```diff
@@ -18,13 +19,97 @@ interface Props {
 
 export default function RequestModal({ show, handleCloseModal, request }: Props) {
   const [loading, setIsLoading] = useState(false);
+  const [validated, setValidated] = useState(false);
+  const [showForm, setShowForm] = useState(false);
   const revalidator = useRevalidator();
+  const { id: neighborhoodId } = useParams();
+  const submit = useSubmit();
+
+  const user = localStorage.getItem('user');
+  const { username, id: userId } = user ? JSON.parse(user) : null;
+
+  const validateTextArea = (event: FocusEvent<HTMLTextAreaElement>) => {
+    const textarea = event.currentTarget as HTMLTextAreaElement;
+    const validPattern = /\s*(\S\s*){4,}/;
+
+    if (!validPattern.test(textarea.value)) {
+      textarea.setCustomValidity('The content needs to be at least 4 characters long.');
+    } else {
+      textarea.setCustomValidity('');
+    }
+
+    textarea.reportValidity();
+  };
+
+  const hideValidation = (event: ChangeEvent<HTMLTextAreaElement>) =>
+    event.currentTarget.setCustomValidity('');
+
+  const handleResponseSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const form = event.currentTarget;
+    if (!form.checkValidity()) {
+      event.stopPropagation();
+
+      setValidated(true);
+    } else {
+      submit(form, {
+        method: 'post',
+        action: `/neighborhoods/${neighborhoodId}`,
+      });
+      setShowForm(false);
```

Comment:
That's what I was thinking as a measure to prevent users from spamming requests.
I see requests & responses as a starting point, conversations should be taken to e-mail or a private chat system.

Reply:
OK, that makes sense. That brings up another issue for another day: we need to be able to edit responses.